### ISSUE TITLE
some about "BatchNorm"

### ISSUE THREAD

**Issue by luweishuang** [dmlc/mxnet#3320]
Recently I do some job in converting torch model to mxnet model and i stucked in mxnet's BatchNorm layer.  In torch model my "nn.SpatialBatchNormalization()" doing batchnormalization in a manner like use_global_stats=False in caffe. But in mxnet, I found I can't do the same because predict() do  " "is_train=False" twice which leads BatchNorm layer always doing "moving average and moving var" other than doing current minibatch mean and var, here is my test code:

######################code_start########################
import find_mxnet  
import mxnet as mx  
import numpy as np

def get_Symbol():
    bn_momentum=0.1
    input_data = mx.symbol.Variable(name="data")
    bn1 = mx.symbol.BatchNorm(name= 'bn1', data=input_data, fix_gamma=False,  momentum=bn_momentum, eps=1e-5, use_global_stats=False)
    return bn1

def creat_MxnetModel(mxnetModelPrefix):
    net_part = get_Symbol()     
    input_dim = [1, 5, 4, 4]
    part_shapes, output_shapes, aux_shapes = net_part.infer_shape(data=tuple(input_dim))
    
    part_params_dic = {}
    part_params_dic['bn1_gamma'] = mx.nd.zeros((5L,))
    part_params_dic['bn1_gamma'][:] = 1.0
    part_params_dic['bn1_beta'] = mx.nd.zeros(( 5L,))
    part_params_dic['bn1_beta'][:] = 0.0

    part_auxStates_dic = {}
    part_auxStates_dic['bn1_moving_mean'] = mx.nd.zeros((5L,))
    part_auxStates_dic['bn1_moving_mean'][:] =0.0
    part_auxStates_dic['bn1_moving_var'] = mx.nd.zeros((5L,))
    part_auxStates_dic['bn1_moving_var'][:] = 1.0

    model_part = mx.model.FeedForward(ctx=mx.cpu(), symbol=net_part, arg_params=part_params_dic, aux_params=part_auxStates_dic, num_epoch=1, learning_rate=0.05, momentum=0.9, wd=0.0001)   
    model_part.save(mxnetModelPrefix)
    print 'creat MxnetModel  done'

if __name__ == '__main__':
    mxnetModelPrefix = 'tmpSave/creatModels'
    creat_MxnetModel(mxnetModelPrefix)
    
    arr = np.zeros([1, 5, 4, 4])
    arr[0,0,:,:] = np.array([[1, 2,3,4],[5,6,7,8], [1, 2,3,4],  [5,6,7,8]])
    arr[0,1,:,:] = np.array([[2,2,2,2],[2,2,2,2], [2,2,2,2],  [2,2,2,2]])
    arr[0,2,:,:] = np.array([[1, 1,1,1],[2,2,2,2], [3,3,3,3],  [4,4,4,4]])
    arr[0,3,:,:] = np.array([[1, 2,1,2],[2,2,2,2], [3,4,3,4],  [4,4,4,4]])
    arr[0,4,:,:] = np.array([[1, 1,1,1],[1,1,1,1], [3,3,3,3],  [4,4,4,4]])
    
    mxnetModel = mx.model.FeedForward.load(mxnetModelPrefix, 1, ctx=mx.cpu(), numpy_batch_size=1,is_train=True)  
    part_params_dic = mxnetModel.arg_params
    part_auxStates_dic = mxnetModel.aux_params       
    print(part_params_dic['bn1_gamma'].asnumpy())
    print(part_params_dic['bn1_beta'].asnumpy())
    print(part_auxStates_dic['bn1_moving_mean'].asnumpy())
    print(part_auxStates_dic['bn1_moving_var'].asnumpy())
    y = mxnetModel.predict(arr)       
    print("mxnet  forward ")
    print("cur_feature.max: ", y.max())
    print("cur_feature.min: ", y.min())
    print y
##################code_end############################

is there another method which can replace "predict" function and set is_train=True to do forward

**Comment by piiswrong** [comment]
you shouldn't use current batch's statistics during testing. If Torch does it, it's a bug in torch. If for some strange reason you really need this, you should modify the C++ code.
